### PR TITLE
Add env & input validation

### PR DESCRIPTION
**Pull request by Naoe-Adrian**
Good Day team mates I add validation on transfer feature. On our next sprint I'll try to improve it more on account number. If account number does not exist in our database it will notify the user immediately.

image

image

### CODE REVIEW

File: app/bank_transfer/index.php [hunk 1]
```diff
@@ -59,19 +59,22 @@
                             <p>Account Number</p>
                         </label>
                         <input type="number" placeholder="Enter Account Number" name="acc_num" id="account_number" />
+                        <div class="error-message" id="account_number_error">
+                        </div>
```
Comment (JheanKhendrick) [inline app/bank_transfer/index.php:63]:
```suggestion
```

File: app/database/deposit.php [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
+$dbUser = $_ENV['DB_USERNAME'];
+$dbPass = $_ENV['DB_PASSWORD'];
+$dbName = $_ENV['DB_DATABASE'];
+
+$conn = new mysqli($dbHost, $dbUser, $dbPass, $dbName);
+
+    // Check the connection
+    if ($conn->connect_error) {
+        die("Connection failed: " . $conn->connect_error);
+    }
```
Comment (JheanKhendrick) [inline app/database/deposit.php:19]:
```suggestion
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}
```

File: app/database/funds-transfer.php [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+$dbUser = $_ENV['DB_USERNAME'];
+$dbPass = $_ENV['DB_PASSWORD'];
+$dbName = $_ENV['DB_DATABASE'];
+
+$conn = new mysqli($dbHost, $dbUser, $dbPass, $dbName);
+
+    // Check the connection
+    if ($conn->connect_error) {
+        die("Connection failed: " . $conn->connect_error);
+    }
```
Comment (JheanKhendrick) [inline app/database/funds-transfer.php:19]:
```suggestion
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}
```

**Review by JheanKhendrick — CHANGES_REQUESTED** [review]

File: app/database/get-acc-info.php [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
+$dbUser = $_ENV['DB_USERNAME'];
+$dbPass = $_ENV['DB_PASSWORD'];
+$dbName = $_ENV['DB_DATABASE'];
+
+$conn = new mysqli($dbHost, $dbUser, $dbPass, $dbName);
+
+    // Check the connection
+    if ($conn->connect_error) {
+        die("Connection failed: " . $conn->connect_error);
+    }
```
Comment (JheanKhendrick) [inline app/database/get-acc-info.php:19]:
```suggestion
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}
```

File: app/database/send-otp-transfer.php [hunk 5]
```diff
@@ -11,6 +10,23 @@
...
+$dbUser = $_ENV['DB_USERNAME'];
+$dbPass = $_ENV['DB_PASSWORD'];
+$dbName = $_ENV['DB_DATABASE'];
+
+$conn = new mysqli($dbHost, $dbUser, $dbPass, $dbName);
+
+    // Check the connection
+    if ($conn->connect_error) {
+        die("Connection failed: " . $conn->connect_error);
+    }
```
Comment (JheanKhendrick) [inline app/database/send-otp-transfer.php:26]:
```suggestion
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}
```

File: app/database/send-otp-transfer.php [hunk 6]
```diff
@@ -11,6 +10,23 @@
...
+
+
+
```
Comment (JheanKhendrick) [inline app/database/send-otp-transfer.php:29]:
```suggestion

```

File: app/database/transfer.php [hunk 7]
```diff
@@ -1,6 +1,22 @@
...
+$dbUser = $_ENV['DB_USERNAME'];
+$dbPass = $_ENV['DB_PASSWORD'];
+$dbName = $_ENV['DB_DATABASE'];
+
+$conn = new mysqli($dbHost, $dbUser, $dbPass, $dbName);
+
+    // Check the connection
+    if ($conn->connect_error) {
+        die("Connection failed: " . $conn->connect_error);
+    }
```
Comment (JheanKhendrick) [inline app/database/transfer.php:19]:
```suggestion
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}
```

File: app/bank_transfer/bank_transfer.js [hunk 8]
```diff
@@ -102,4 +102,46 @@ function clearForm() {
   document.getElementById('amount').value = '';
   document.getElementById('transfer-type').value = '';
   document.getElementById('bank-selector').value = '';
-}
\ No newline at end of file
+}
+
+var transferTypeSelect;
+var bankSelector;
```
Comment (JheanKhendrick) [inline app/bank_transfer/bank_transfer.js:108]:
```suggestion
let transferTypeSelect;
let bankSelector;
```

**Review by JheanKhendrick — CHANGES_REQUESTED** [review]

**Comment by JheanKhendrick** [comment]
Would it be possible for the frontend validation to behave consistently with the one implemented for the signup feature?

example:
image

**Comment by Naoe-Adrian** [comment]
Thank you, Jhean. I tried your method, but I'm having difficulty adding a border when the user inputs invalid values in the fields. Additionally, I've disabled the dropdown for the bank because users should not change it, the system automatically handles the exchange between internal and external transfers. 
image


**_ON INTERNAL VALUE_**
image

**_ON EXTERNAL VALUE_**
image

**Review by JohnStevenPenecitos — APPROVED** [review]

**Pull request merged**